core/overdrive.py

```diff
@@ -199,7 +259,54 @@ def __init__(self, _db, collection):
         # This is set by an access to .collection_token
         self._collection_token = None
 
-    def endpoint(self, url, **kwargs):
+    def _determine_hosts(self, configuration: OverdriveConfiguration) -> dict[str, str]:
+        # Figure out which hostnames we'll be using when constructing
+        # endpoint URLs.
+        server_nickname = (
+            configuration.server_nickname or OverdriveConfiguration.PRODUCTION_SERVERS
+        )
+        if server_nickname not in self.HOSTS:
```

Comment:
In case of a misconfigured environment, why not fail rather than default to the Production API?

Reply:
Good question... I didn't actually write that code, I just adapted it to the new system. Apparently falling back to the production API was the original intent.